addons/project/models/project.py

```diff
@@ -1368,6 +1369,14 @@ def _get_view(self, view_id=None, view_type='form', **options):
                 node.set('invisible', '1')
         return arch, view
 
+    @api.onchange('stage_id')
+    def _onchange_stage_id(self):
+        self.kanban_state = 'normal'
+
+    @api.onchange('project_id')
+    def _onchange_project_id(self):
+        self.kanban_state = 'normal'
+
```

Comment:
```suggestion
    @api.onchange('stage_id', 'project_id')
    def _onchange_kanban_state(self):
        self.kanban_state = 'normal'

```

Reply:
same question

Reply:
Name it the way you want but duplicating the code is just a bad practice so  merge them together as proposed.